### PR TITLE
fix(ts/analyzer): remove unused code in the class for simplifying Union Type

### LINKED ISSUE #175: Append `undefined` to type list on optional member access
https://github.com/dudykr/stc/blob/e69d14c1b30473f73bc8873400e950d44a047e64/crates/stc_ts_file_analyzer/tests/pass/exprs/optionalChaining/.access-6.ts#L2-L3

The type of `o6<number>()?.["x"];` above should be `number | undefined`, but currently it's `number`.

Relevant code for optional chaining is  https://github.com/dudykr/stc/blob/e69d14c1b30473f73bc8873400e950d44a047e64/crates/stc_ts_file_analyzer/src/analyzer/expr/optional_chaining.rs#L25-L49

### PR DESCRIPTION
**Pull request by nissy-dev**
Fix #175 

**Description:**

I remove unused code in the class for simplifying Union Type and add the test case.

**BREAKING CHANGE:**

**Related issue (if exists):**

### CODE REVIEW

**Comment by kdy1** [comment]
Van you run check.sh to update stats?

File: crates/stc_ts_file_analyzer/src/analyzer/generalize.rs [hunk 1]
```diff
@@ -72,7 +72,6 @@ impl Simplifier<'_> {
 impl Fold<Union> for Simplifier<'_> {
     fn fold(&mut self, mut union: Union) -> Union {
         let should_remove_null_and_undefined = union.types.iter().any(|ty| match ty.normalize() {
-            Type::TypeLit(..) => true,
```
Comment (nissy-dev) [inline crates/stc_ts_file_analyzer/src/analyzer/generalize.rs:75]: Currently , I removed only this line. If I can remove `should_remove_null_and_undefined` itself, I will add new commit.

**Comment by kdy1** [comment]
You have to run it again

**Comment by nissy-dev** [comment]
> You have to run it again, also, can you run it after rebasing?

Yes. there is no additional differences.

File: crates/stc_ts_file_analyzer/src/analyzer/generalize.rs (hunk 1 above)
Comment (kdy1) [reply crates/stc_ts_file_analyzer/src/analyzer/generalize.rs:75]: I think we should remove all of them

Comment (kdy1) [reply crates/stc_ts_file_analyzer/src/analyzer/generalize.rs:75]: Can you remove all of this?

**Review by kdy1 — CHANGES_REQUESTED** [review]

File: crates/stc_ts_file_analyzer/src/analyzer/generalize.rs (hunk 1 above)
Comment (nissy-dev) [reply crates/stc_ts_file_analyzer/src/analyzer/generalize.rs:75]: I removed!

**Review by kdy1 — APPROVED** [review]

**Comment by kdy1** [comment]
Thanks!

**Pull request merged**